cluster-api/cloud/google/machinesetup/config_types.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machinesetup
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"reflect"
+
+	"github.com/ghodss/yaml"
+	clustercommon "k8s.io/kube-deploy/cluster-api/pkg/apis/cluster/common"
+	clusterv1 "k8s.io/kube-deploy/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+type ConfigWatch struct {
+	path string
+}
+
+type ValidConfigs struct {
+	configList *configList
+}
+
+type configList struct {
+	Items []config `json:"items"`
+}
+
+type config struct {
+	// A list of the valid combinations of ConfigParams that will
+	// map to the given Image and Metadata.
+	Params []ConfigParams `json:"machineParams"`
+
+	// The fully specified image path.
```

Comment:
Does that include `https://www.googleapis.com/compute/beta/projects/ubuntu-os-cloud/global/images/ubuntu-1604-xenial` along with `/projects/ubuntu-os-cloud/global/images/ubuntu-1604-xenial-v20180405`?